app/api/log/route.ts

```
@@ -0,0 +1,47 @@
+    }
+
+    try {
+        const body = await req.json();
+        const { message, data } = body;
+
+        if (!message || typeof message !== 'string') {
+            return NextResponse.json({ success: false, error: 'Invalid message format' }, { status: 400 });
+        }
+
+        // Shared constant for control character regex using Unicode escapes
+        const CONTROL_CHARS = /[\u0000-\u001F\u007F]/g;
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)